### PR TITLE
Require access-control write protection for first deployments.

### PR DESCRIPTION
**Pull request by gjoranv**
Refer to the commit message for details.

The test will be expanded, and the TODO will be resolved in later commits.

@hmusum Please review the general concept of first deployment validators.

### CODE REVIEW

**Review by bjorncs — APPROVED** [review]

File: config-model/src/main/java/com/yahoo/vespa/model/application/validation/first/AccessControlValidator.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    @Override
+    public void validate(VespaModel model, DeployState deployState) {
+
+        // TODO: what about zone application? (HOSTED_INFRASTRUCTURE)
+
+        if (deployState.isHosted() && deployState.zone().environment().isProduction()) {
+            List<String> offendingClusters = new ArrayList<>();
+            for (ContainerCluster cluster : model.getContainerClusters().values()) {
+                if (cluster.getHttp() == null
```
Comment (hmusum) [inline config-model/src/main/java/com/yahoo/vespa/model/application/validation/first/AccessControlValidator.java]: When is this the case?

**Review by hmusum — APPROVED** [review]
It could be useful with some sort of switch if this does not work as expected

File: config-model/src/main/java/com/yahoo/vespa/model/application/validation/first/AccessControlValidator.java (hunk 1 above)
Comment (gjoranv) [reply config-model/src/main/java/com/yahoo/vespa/model/application/validation/first/AccessControlValidator.java]: When there is no 'http' element in services.xml. It's covered in the unit test `cluster_with_handler_fails_validation_without_http_element`

**Comment by gjoranv** [comment]
@hmusum Is there a pattern for such a switch?

**Comment by hmusum** [comment]
@gjoranv Unfortunately, no. There have been cases where adding a new field to 'configserver' config with a default has worked OK (configserver-config.xml can be added to configserver-app directory and will override defaults if needed). But you will have to propagate this all the way down to where you need it, so it will be some work.

**Comment by gjoranv** [comment]
@bjorncs, @hmusum Please take a look at the two last commits.

@tokle Please review. I think all corner cases are covered now.

**Comment by hmusum** [comment]
Last two commits LGTM

**Review by tokle — APPROVED** [review]

**Comment by tokle** [comment]
Consider adding a "feature" flag to make this available in CD system only.

**Comment by hmusum** [comment]
👍

**Review by bjorncs — APPROVED** [review]

**Pull request merged**